### ISSUE TITLE
Allow user to pass custom search paths

### ISSUE THREAD

**Issue by flferretti** [ami-iit/resolve-robotics-uri-py#12]
When using [`robot_descriptions`](https://github.com/robot-descriptions/robot_descriptions.py), this library cannot resolve the URIs due to the fact that `robot_descriptions` does not expose any environment variable of the one supported:

```python
SupportedEnvVars = {
    "AMENT_PREFIX_PATH",
    "GAZEBO_MODEL_PATH",
    "GZ_SIM_RESOURCE_PATH",
    "IGN_GAZEBO_RESOURCE_PATH",
    "ROS_PACKAGE_PATH",
    "SDF_PATH",
}
```

As a result, `resolve_robotics_uri_py` cannot locate the necessary resources. Do you think that it could be beneficial to allow passing an additional environment variable for specifying custom search path as in https://github.com/ami-iit/resolve-robotics-uri-py/commit/1c9e78f8b4164f2710ec2dd3d123153665cd0328?

C.C. @traversaro @lorycontixd

**Comment by traversaro** [comment]
Good point, this is indeed something supported by many libraries, see https://github.com/RobotLocomotion/drake-external-examples/issues/170 .

> Do you think that it could be beneficial to allow passing an additional environment variable for specifying custom search path as in [1c9e78f](https://github.com/ami-iit/resolve-robotics-uri-py/commit/1c9e78f8b4164f2710ec2dd3d123153665cd0328)?

In which sense "an additional environment variable" ? I guess you want an additional argument to the function, not an env variable?

However, I am a bit confused how existing users of `robot_descriptions` can resolve `package://` URIs if no env variables are set.

**Comment by traversaro** [comment]
> However, I am a bit confused how existing users of `robot_descriptions` can resolve `package://` URIs if no env variables are set.

Mistery solved, the library-specific loader handled that (see https://github.com/robot-descriptions/robot_descriptions.py/blob/d0dc260953a459450e73d9a4b456fd0b02ace59d/robot_descriptions/loaders/idyntree.py#L54C39-L54C55 for example with iDynTree). Then I guess it totally make sense to have this functionality here, can you open a PR so it is easier to discuss the change?

**Comment by flferretti** [comment]
> Good point, this is indeed something supported by many libraries, see [RobotLocomotion/drake-external-examples#170](https://github.com/RobotLocomotion/drake-external-examples/issues/170) .
> 
> > Do you think that it could be beneficial to allow passing an additional environment variable for specifying custom search path as in [1c9e78f](https://github.com/ami-iit/resolve-robotics-uri-py/commit/1c9e78f8b4164f2710ec2dd3d123153665cd0328)?
> 
> In which sense "an additional environment variable" ? I guess you want an additional argument to the function, not an env variable?
> 
> However, I am a bit confused how existing users of `robot_descriptions` can resolve `package://` URIs if no env variables are set.

Yes, sorry, I meant an extra argument that represents an environment variable associated with an additional search path

**Comment by flferretti** [comment]
Closed by #13
